src/vellum/workflows/runner/runner.py

```diff
@@ -250,6 +258,105 @@ def __init__(
+        - Preserves backward compatibility for existing code
+        """
+        # Collect all IntegrationTrigger types declared in the workflow
+        workflow_integration_triggers = []
+        for subgraph in self.workflow.get_subgraphs():
+            for trigger_type in subgraph.triggers:
+                if issubclass(trigger_type, IntegrationTrigger):
+                    workflow_integration_triggers.append(trigger_type)
+
+        # Case 1: Trigger instance provided - validate and bind
+        if trigger:
+            if not workflow_integration_triggers:
+                raise WorkflowInitializationException(
+                    message="trigger provided but workflow does not have an IntegrationTrigger",
+                    workflow_definition=self.workflow.__class__,
+                    code=WorkflowErrorCode.INVALID_INPUTS,
+                )
+
+            # Validate that the trigger instance is compatible with one of the workflow's trigger types
+            # Allow subclasses: if workflow declares BaseSlackTrigger, accept SpecificSlackTrigger instances
+            trigger_class = type(trigger)
+
+            # Check if the provided trigger_class is a subclass of any declared trigger type
+            if not any(
+                issubclass(trigger_class, declared_trigger) for declared_trigger in workflow_integration_triggers
+            ):
+                raise WorkflowInitializationException(
+                    message=f"Provided trigger type {trigger_class.__name__} is not compatible with workflow triggers. "
+                    f"Workflow has: {[t.__name__ for t in workflow_integration_triggers]}",
+                    workflow_definition=self.workflow.__class__,
+                    code=WorkflowErrorCode.INVALID_INPUTS,
+                )
+
+            # Bind trigger to state
+            trigger.bind_to_state(self._initial_state)
+
+            # Filter entrypoints to the SPECIFIC trigger subclass path only
+            # UNLESS caller explicitly provided entrypoints (e.g., via entrypoint_nodes or external_inputs)
+            if not self._caller_provided_entrypoints:
+                # Use the specific trigger subclass, not the parent IntegrationTrigger
+                specific_entrypoints = self._get_entrypoints_for_trigger_type(trigger_class)
+                if specific_entrypoints:
+                    self._entrypoints = specific_entrypoints
+
+        # Case 2: No trigger but workflow has IntegrationTrigger(s)
+        elif workflow_integration_triggers:
+            if not self._has_manual_trigger():
+                # Workflow has ONLY IntegrationTrigger - this is an error
+                raise WorkflowInitializationException(
+                    message="Workflow has IntegrationTrigger which requires trigger parameter",
+                    workflow_definition=self.workflow.__class__,
+                    code=WorkflowErrorCode.INVALID_INPUTS,
+                )
```

Comment:
[oos] This case IIUC can be:
```
graph = {
    SlackTrigger >> TopNode,
    BottomNode,
}
```

In this kind of graph, I believe we still want to consider `BottomNode` as being "Manually Triggered".